vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/treegrid/demo/TreeGridDemo.java

```diff
@@ -146,4 +149,78 @@ public boolean hasChildren(Account item) {
 
         addCard("TreeGrid with lazy loading", grid);
     }
+
+    // Drag and drop
+    private Department draggedItem;
+
+    private void createDragAndDrop() {
+        DepartmentData departmentData = new DepartmentData();
+        // begin-source-example
+        // source-example-heading: Drag and drop
+        TreeGrid<Department> treeGrid = new TreeGrid<>();
+
+        treeGrid.setDataProvider(
+                new AbstractHierarchicalDataProvider<Department, Object>() {
+                    @Override
+                    public int getChildCount(
+                            HierarchicalQuery<Department, Object> query) {
+                        return departmentData.getChildCount(query.getParent());
+                    }
+
+                    @Override
+                    public Stream<Department> fetchChildren(
+                            HierarchicalQuery<Department, Object> query) {
+                        return departmentData
+                                .getChildDepartments(query.getParent())
+                                .stream();
+                    }
+
+                    @Override
+                    public boolean hasChildren(Department item) {
+                        return departmentData.hasChildren(item);
+                    }
+
+                    @Override
+                    public boolean isInMemory() {
+                        return false;
+                    }
+                });
+
+        treeGrid.addHierarchyColumn(Department::getName)
+                .setHeader("Department Name");
+        treeGrid.addColumn(Department::getManager).setHeader("Manager");
+        treeGrid.setRowsDraggable(true);
+        treeGrid.setDropMode(GridDropMode.ON_TOP);
+
+        treeGrid.addDragStartListener(
+                event -> draggedItem = event.getDraggedItems().get(0));
+
+        treeGrid.addDragEndListener(event -> draggedItem = null);
```

Comment:
draggedItem should be part of the code shown to the user

Reply:
Done.